sdk/core/src/services/CognitoService.ts

```diff
@@ -278,11 +278,8 @@ export default class CognitoService {
 
     const { clientId: ClientId } = this.cognitoOptions
 
-    const params = { ClientId, Username }
-
-    if (messageParameters) {
-      Object.assign(params, { ClientMetadata: messageParameters })
-    }
+    const ClientMetadata = messageParameters || {}
+    const params = { ClientId, Username, ClientMetadata }
```

Comment:
@VitaliiShpital did you run all tests and their working as expected for all custom message use cases, correct ?
Wondering if nothing get broken when ClientMetadata present and empty (just thinking why it was written this way before)

Reply:
Yeah nothing have changed with this tweak. I've actually copy-pasted this from other method in this class.